flow/environments/xsede.py

```diff
@@ -215,8 +215,67 @@ def add_args(cls, parser):
         )
 
 
+class ExpanseEnvironment(DefaultSlurmEnvironment):
+    """Environment profile for the Expanse supercomputer.
+
+    https://www.sdsc.edu/support/user_guides/expanse.html
+    """
+
+    hostname_pattern = r".*\.expanse\.sdsc\.edu$"
+    template = "expanse.sh"
+    cores_per_node = 128
+    gpus_per_node = 4
+
+    @classmethod
+    def _get_mpi_prefix(cls, operation, parallel):
+        # The Expanse user guide recommends the usage of `srun`. However, this will not
+        # forward the MPI configuration to the container, meaning it cannot be used with
+        # one of flow's most common use cases, HPC with singularity containers. Thus we
+        # are using the default mpi command `mpiexec` with the other recommendations
+        # given by the Expanse user guide. Native MPI programs with Expanse also require
+        # the `--mpi=pmi2` option (or the documentation suggests as such), but this
```

Comment:
In my understanding, the flag `--mpi=pmi2` may depend on the MPI module being used. The docs list "OpenMPI, MVAPICH2, and IntelMPI" and my guess is that `pmi2` corresponds to MVAPICH2. @joaander may know more.
https://www.sdsc.edu/support/user_guides/expanse.html#compiling